cmd/client_gnfd.go

```diff
@@ -46,29 +26,27 @@ func NewClient(ctx *cli.Context) (*gnfdclient.GnfdClient, error) {
 
 	privateKeyStr := ctx.String("privateKey")
 	if privateKeyStr == "" {
-		// generate private key if not provided
-		privKey, _, _ := testdata.KeyEthSecp256k1TestPubAddr()
-		privateKeyStr = hex.EncodeToString(privKey.Bytes())
-	}
-
-	keyManager, err := keys.NewPrivateKeyManager(privateKeyStr)
-	if err != nil {
-		log.Error().Msg("fail to new key manager" + err.Error())
+		return nil, fmt.Errorf("failed to private key")
 	}
 
-	client, err := gnfdclient.NewGnfdClient(grpcAddr, chainId, endpoint, keyManager, false,
-		WithKeyManager(keyManager),
-		WithGrpcDialOption(grpc.WithTransportCredentials(insecure.NewCredentials())))
+	account, err := sdkTypes.NewAccountFromPrivateKey("gnfd-account", privateKeyStr)
 	if err != nil {
-		fmt.Println("failed to create client" + err.Error())
+		return nil, err
 	}
 
+	var cli client.Client
 	host := ctx.String("host")
 	if host != "" {
-		client.SPClient.SetHost(host)
+		cli, err = client.New(chainId, grpcAddr, client.Option{DefaultAccount: account, Host: host})
+	} else {
+		cli, err = client.New(chainId, grpcAddr, client.Option{DefaultAccount: account})
+	}
+
+	if err != nil {
+		fmt.Println("failed to create client" + err.Error())
```

Comment:
can use the fmt.Printf().. for others as well

Reply:
updated